src/main/kotlin/com/codingchallenge/transactionanalyser/TransactionProcessor.kt

```diff
@@ -0,0 +1,24 @@
+package com.codingchallenge.transactionanalyser
+
+import com.codingchallenge.transactionanalyser.domain.Transaction
+import com.codingchallenge.transactionanalyser.domain.UserInput
+import java.math.BigDecimal
+
+class TransactionProcessor {
+    fun filterTransactionsByUserInput(transactions: List<Transaction>, input: UserInput): List<Transaction>{
+        return transactions
+            // find the transactions related to the account
+            .filter { it.fromAccountId == input.accountId || it.toAccountId == input.accountId }
+            // find the transactions where the createdAt is after the "from" time and before the "to" time
+            .filter { it.createdAt > input.from && it.createdAt < input.to }
+            // Remove transactions that get reversed
+            .filter { ft -> transactions.none { t -> ft.transactionId == t.relatedTransaction } }
```

Comment:
rather than using `ft` consider naming it to what it actually is. Naming it `ft` makes the reader guess.